### PR TITLE
jit(arm64): restore callee-saved registers on exits from native code 

### LINKED ISSUE #308: Default JIT when arm64 or amd64
When we sort out CI, let's set JIT as default.

### PR DESCRIPTION
**Pull request by mathetake**
fixes #308

### CODE REVIEW

File: config_supported.go [hunk 1]
```diff
@@ -6,5 +6,5 @@ package wazero
 // NewRuntimeConfig returns NewRuntimeConfigInterpreter
 // TODO: switch back to NewRuntimeConfigJIT https://github.com/tetratelabs/wazero/issues/308
```
Comment (codefromthecrypt) [inline config_supported.go:7]:
```suggestion
```

**Review by codefromthecrypt — COMMENTED** [review]
When ready... :D

**Review by codefromthecrypt — APPROVED** [review]
Wow keen eye fixing this one!

File: examples/testdata/host_func.go [hunk 2]
```diff
@@ -42,6 +42,6 @@ func base64OnString(num uint32) {
 	for i := uint32(0); i < num; i++ {
 		msg := getRandomString()
 		encoded := base64.StdEncoding.EncodeToString([]byte(msg))
-		fmt.Printf("base64 encoded string '%s'\n", encoded[:10])
```
Comment (codefromthecrypt) [inline examples/testdata/host_func.go:45]: anything interesting here? except maybe the massive reduction in wasm :D

Comment (mathetake) [reply examples/testdata/host_func.go:45]: oh nothing special but funny that after rebuild amd64 has started failing on this due to "out of range" so I just deleted :D

Comment (mathetake) [reply examples/testdata/host_func.go:45]: it seems like a TinyGo's bug as I see this succeeds after changing this to 
```
		v := encoded[:10]
		fmt.Printf("base64 encoded string '%s'\n", v)
```

Comment (mathetake) [reply examples/testdata/host_func.go:45]: something buggy on inline slicing :D

**Pull request merged**